edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Service/Program.cs

```diff
@@ -433,8 +437,50 @@ static Task Cleanup(Option<Agent> agentOption, ILogger logger)
                 logger.LogError(AgentEventIds.Agent, ex, "Error on shutdown");
                 return Task.CompletedTask;
             }
-        }
-
+        }
+
+        static void ValidateStorageIdentity(string storagePath, string deviceId, string iotHubHostname, string moduleId, ILogger logger, string moduleGenerationId = "")
+        {

```

Comment:
We should create a module specific folder inside the storage folder, and put all data, including the file in there. That way, if mismatch found, we can just blow the folder itself and re-create it.

Reply:
are you doing this? If not, can you please take care of it?

Reply:
Hey Varun, so inside the storage folder, there are already module specific folders created ( which is also why now I am deleting and recreating the folder instead of going through and deleting everything one by one)

Reply:
Which is why I was a bit confused about the first comment: what that logic was doing was going through the respective subfolder for the module, deleting all files in that folder, and then creating the Device Identity file